### ISSUE TITLE
Use AngularJS commit message conventions

### ISSUE THREAD

**Issue by JamieMagee** [dependabot/dependabot-core#246]
Using [AngularJS commit message conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) for dependabot would allow for automatic changelog creation using [coventional-changelog](https://github.com/conventional-changelog/conventional-changelog) (and other associated tools).

For dependabot commits, this would look something like

```
chore(node): bump #{dependency.name} from #{previous_version(dependency)} to {new_version(dependency)}

CURRENT COMMIT DETAIL GOES HERE
```

I am able to make the changes required in `message_builder.rb`, but I would like to hear your opinion on my suggestion before I do any work.

**Comment by greysteil** [comment]
Thanks for the feedback. Is the problem that Dependabot is creating `build: ...` commit messages for you rather than `chore...` ones, or is it not creating semantic commits at all?

**Comment by JamieMagee** [comment]
The issue is more that it's not creating semantic commits at all. The AngularJS commit message conventions show what form a semantic commit message should take:

```
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
```

The current dependabot commit messages don't conform to this standard.

I think that following these conventions would allow projects that already follow them to make use of dependabot, without extra manual intervention for commits/PRs.

**Comment by greysteil** [comment]
Hmmm, that’s odd - Dependabot should be autodetecting this. Can you ping me the full name of the repo it’s not doing so on? support@dependabot.com <mailto:support@dependabot.com> if you don’t want to put the name here.

(I’m on a boat in the middle of the pacific, so won’t be able to debug this as fast as normal. Also, my emails / comments might lack the most recent context, as I’m only able to download / send them once a day.)

**Comment by JamieMagee** [comment]
Sure, the repo is https://github.com/JamieMagee/hashflags-node. I just had another look at the code in `message_builder.rb`, and I can see that it checks if there are any recent commits with semantic prefixes. I recently switched over to using semantic prefixes in this repo, so maybe there was some issue/caching with getting the latest commit messages?

I'm in no hurry, enjoy your trip!

**Comment by greysteil** [comment]
I’ve just pushed a small change that should make Dependabot a bit quicker to switch over when you change to using semantic prefixes - it now excludes its own commits when analysing what conventions a repo is using.

I’m pretty sure that the above will fix the issue for you, and mean #254 (which would affect users who aren’t using semantic commit messages) isn’t required. Can you let me know based on the PRs you receive next week?

Thanks again for the feedback, and your patience on this one!

**Comment by JamieMagee** [comment]
I took a closer look at one of the [PR failures](https://travis-ci.org/JamieMagee/hashflags-node/jobs/339007862#L497) and the failure message is:

```
input: build: Bump @types/node from 9.4.1 to 9.4.2
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
```

It's [this rule](https://github.com/marionebl/commitlint/tree/master/@commitlint/config-angular#subject-case) from commitlint that's failing it. My PR #254 would fix it, but as you mentioned, it would also affect users not using semantic commit messages.

I can see three options:

1. dependabot can make the subject start with a lower-case letter for all commits.
2. Add some more logic to `message_builder.rb` to change the case of the first letter depending if semantic messages are used are not.
3. I can disable `commitlint` during my builds (but this wouldn't fix it for everyone else)

Personally I think 1. is the easiest, and is solved by #254. What do you think?

**Comment by greysteil** [comment]
Thanks for the detailed debug info! I've implemented (2) [here](https://github.com/dependabot/dependabot-core/commit/d969cf91429567e578896ed885fc62ba6318aeca) and just released it - all future semantic commit message PRs will use that convention. 🎉 

(I ended up going for (2) rather than merging your PR on (1) because I really care about Dependabot doing things as close to perfectly as possible, at the expense of a more complicated codebase. Dependabot's scope is really limited (dependency update PRs) so I think we can do it perfectly.)

Thanks again!
